src/functionBased/components/TodoContainer.js

```diff
@@ -0,0 +1,110 @@
+import React,{useState, useEffect} from "react"
+import InputTodo from "./InputTodo"
+import TodoList  from "./TodosList";
+import Header from "./Header";
+import { v4 as uuidv4 } from "uuid";
+import { Route, Switch} from "react-router-dom";
+import About from "../pages/About"
+import NotMatch from "../pages/NotMatch"
+import Navbar from "./Navbar";
+
+const TodoContainer = () => {
+  function getInitialTodos() {
+    // getting stored items
+    const temp = localStorage.getItem("todos")
+    const savedTodos = JSON.parse(temp)
+    return savedTodos || []
+  }
+
+  const [todos, setTodos] = useState(getInitialTodos())
+
+  const handleChange = id => {
+    setTodos(prevState =>
+      prevState.map(todo => {
+        if (todo.id === id) {
+          return {
+            ...todo,
+            completed: !todo.completed,
+          }
+        }
+        return todo
+      })
+    )
+  }
+
+  const delTodo = id => {
+    setTodos([
+      ...todos.filter(todo => {
+        return todo.id !== id
+      }),
+    ])
+  }
+
+  const addTodoItem = title => {
+    const newTodo = {
+      id: uuidv4(),
+      title: title,
+      completed: false,
+    }
+    setTodos([...todos, newTodo])
+  }
+
+  const setUpdate = (updatedTitle, id) => {
+    setTodos(
+      todos.map(todo => {
+        if (todo.id === id) {
+          todo.title = updatedTitle
+        }
+        return todo
+      })
+    )
+  }
+
+  useEffect(() => {
+      // getting stored items
+    const temp = localStorage.getItem("todos")
+    const loadedTodos = JSON.parse(temp)
+
+    if (loadedTodos) {
+      setTodos(loadedTodos)
+    }
+  },[])
+
+  useEffect(() => {
+    // storing todos items
+    const temp = JSON.stringify(todos)
+    localStorage.setItem("todos", temp)
+  }, [todos])
+
+  return (
+    <React.Fragment>
+      <Navbar/>
+      <Switch>
+        <Route exact path="/">
+      <div className="container">
+        <div className="inner">
+          <Header />
+          <InputTodo addTodoProps={addTodoItem} />
+          <TodoList
+            todos={todos}
+            handleChangeProps={handleChange}
+            deleteTodoProps={delTodo}
+            setUpdate={setUpdate}
+         />
+       </div>
+      </div>
+        </Route>
+
+        <Route path="/about" component={About}>
+          <About/>
+        </Route>
+
+        <Route path="*">
+          <NotMatch/>
+        </Route>
+      </Switch>
```

Comment:
This syntax is part of the old version of react-router-dom library v-5.  Please consider referring to the [new syntax](https://reactrouter.com/docs/en/v6/getting-started/overview) of v-6. 
It would be something like this: 
```
<Routes>
    <Route
        path='/'
        element={
          <div className='container'>
          <div className='inner'>
             <Header />
             <InputToDo addTodoProps={addTodoItem} />
             <TodosList
                 todos={todos}
                 handleChangeProps={handleChange}
                 deleteTodoProps={delTodo}
                 setUpdate={setUpdate}
             />
          </div>
          </div>
        }
    />
    <Route path='/about' element={<About />}/>
    <Route path='*' element={<NotMatch />} />
</Routes>
```